code/game/objects/items/surgery_tray.dm

```diff
@@ -0,0 +1,338 @@
+				. += "scalpel_advanced"
+			if(/obj/item/scalpel/cruel)
+				. += "scalpel_cruel"
+			else
+				. += "scalpel_normal"
+	if(cautery)
+		switch(cautery.type)
+			if(/obj/item/cautery/alien)
+				. += "cautery_alien"
+			if(/obj/item/cautery/advanced)
+				. += "cautery_advanced"
+			if(/obj/item/cautery/cruel)
+				. += "cautery_cruel"
+			else
+				. += "cautery_normal"
+	if(hemostat)
+		switch(hemostat.type)
+			if(/obj/item/hemostat/alien)
+				. += "hemostat_alien"
+			if(/obj/item/hemostat/cruel)
+				. += "hemostat_cruel"
+			else
+				. += "hemostat_normal"
+	if(retractor)
+		switch(retractor.type)
+			if(/obj/item/retractor/alien)
+				. += "retractor_alien"
+			if(/obj/item/retractor/advanced)
+				. += "retractor_advanced"
+			if(/obj/item/retractor/cruel)
+				. += "retractor_cruel"
+			else
+				. += "retractor_normal"
+	if(surgical_drill)
+		switch(surgical_drill.type)
+			if(/obj/item/surgicaldrill/alien)
+				. += "drill_alien"
+			else
+				. += "drill_normal"
+	if(circular_saw)
+		switch(circular_saw.type)
+			if(/obj/item/circular_saw/alien)
+				. += "saw_alien"
+			else
+				. += "saw_normal"
+
+/obj/item/surgery_tray/Entered(obj/item/arrived, atom/old_loc, list/atom/old_locs)
+	. = ..()
+	if(istype(arrived, /obj/item/blood_filter))
+		if(blood_filter && blood_filter.toolspeed <= arrived.toolspeed)
+			return
+		blood_filter = arrived
+	else if(istype(arrived, /obj/item/cautery))
+		if(cautery && cautery.toolspeed <= arrived.toolspeed)
+			return
+		cautery = arrived
+	else if(istype(arrived, /obj/item/circular_saw))
+		if(circular_saw && circular_saw.toolspeed <= arrived.toolspeed)
+			return
+		circular_saw = arrived
+	else if(istype(arrived, /obj/item/hemostat))
+		if(hemostat && hemostat.toolspeed <= arrived.toolspeed)
+			return
+		hemostat = arrived
+	else if(istype(arrived, /obj/item/surgicaldrill))
+		if(surgical_drill && surgical_drill.toolspeed <= arrived.toolspeed)
+			return
+		surgical_drill = arrived
+	else if(istype(arrived, /obj/item/retractor))
+		if(retractor && retractor.toolspeed <= arrived.toolspeed)
+			return
+		retractor = arrived
+	else if(istype(arrived, /obj/item/scalpel))
+		if(scalpel && scalpel.toolspeed <= arrived.toolspeed)
+			return
+		scalpel = arrived
+	else if(istype(arrived, /obj/item/bonesetter))
+		if(bonesetter)
+			return
+		bonesetter = arrived
+	else if(istype(arrived, /obj/item/razor))
+		if(razor)
+			return
+		razor = arrived
+	else if(istype(arrived, /obj/item/stack/medical/bone_gel))
+		if(bone_gel)
+			return
+		bone_gel = arrived
+	else if(istype(arrived, /obj/item/stack/sticky_tape/surgical))
+		if(surgical_tape)
+			return
+		surgical_tape = arrived
+	else if(istype(arrived, /obj/item/surgical_drapes))
+		if(surgical_drapes)
+			return
+		surgical_drapes = arrived
+
+/obj/item/surgery_tray/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == blood_filter)
+		var/obj/item/blood_filter/other_filter = locate() in contents
+		blood_filter = other_filter || null
```

Comment:
All of these `|| null`s are pointless, `locate()` will give you `null` if it doesn't find anything, so at the end of the day you're doing `null || null`

Reply:
Actually, ultimately, you can skip out the middle man because you typed all of these vars, and just do 
a locate: `blood_filter = locate() in contents` (or `locate() in src`, it's the same thing. I prefer the latter but it's just preference)